### PR TITLE
Add support for CustomCollection API

### PR DESCRIPTION
**Pull request by jeremy-parr**
PR adds support for the CustomCollection APIs from shopify.

[https://help.shopify.com/api/reference/customcollection](https://help.shopify.com/api/reference/customcollection)

### CODE REVIEW

File: customcollection.go [hunk 1]
```diff
@@ -0,0 +1,96 @@
...
+	path := fmt.Sprintf("%s.json", customCollectionsBasePath)
+	wrappedData := CustomCollectionResource{Collection: &collection}
+	resource := new(CustomCollectionResource)
+	err := s.client.Post(path, wrappedData, resource)
+	return resource.Collection, err
+}
+
+// Update an existing custom collection
+func (s *CustomCollectionServiceOp) Update(collectionID int, collection CustomCollection) (*CustomCollection, error) {
+	path := fmt.Sprintf("%s/%d.json", customCollectionsBasePath, collectionID)
```
Comment (dlebech) [inline customcollection.go]: Could you change the signature of the `Update` function so it takes only the `CustomCollection` to update as parameter and then uses the `ID` field? See e.g. the [`Product`](https://github.com/getconversio/go-shopify/blob/e461b55bc05f9b6e038e02d2b3fb91e1dc2d6d17/product.go#L101-L102) and [`Variant`](https://github.com/getconversio/go-shopify/blob/e461b55bc05f9b6e038e02d2b3fb91e1dc2d6d17/variant.go#L100-L101) `Update` functions.

File: customcollection.go [hunk 2]
```diff
@@ -0,0 +1,96 @@
...
+// CustomCollection represents a Shopify custom collection.
+type CustomCollection struct {
+	ID             int    `json:"id"`
+	Handle         string `json:"handle"`
+	Title          string `json:"title"`
+	UpdatedAt      string `json:"updated_at"`
+	BodyHTML       string `json:"body_html"`
+	SortOrder      string `json:"sort_order"`
+	TemplateSuffix string `json:"template_suffix"`
+	Image          Image  `json:"image"`
```
Comment (dlebech) [inline customcollection.go]: According to the docs, this is [actually a `string`](https://help.shopify.com/api/reference/customcollection#properties). I haven't worked with this collection in a while, so I can't verify. Do you know what the general return type is? Sometimes the docs are not fully updated.

File: customcollection.go [hunk 3]
```diff
@@ -0,0 +1,96 @@
...
+	Published      bool   `json:"published"`
+	PublishedAt    string `json:"published_at"`
```
Comment (dlebech) [inline customcollection.go]: Same as above: you should be able to use a `*time.Time` type here. It works in most other cases, e.g. for [products](https://github.com/getconversio/go-shopify/blob/e461b55bc05f9b6e038e02d2b3fb91e1dc2d6d17/product.go#L38).

File: customcollection.go [hunk 4]
```diff
@@ -0,0 +1,96 @@
...
+type CustomCollectionServiceOp struct {
+	client *Client
+}
+
+// CustomCollection represents a Shopify custom collection.
+type CustomCollection struct {
+	ID             int    `json:"id"`
+	Handle         string `json:"handle"`
+	Title          string `json:"title"`
+	UpdatedAt      string `json:"updated_at"`
```
Comment (dlebech) [inline customcollection.go]: You should be able to use a `*time.Time` type here. It works in most other cases, e.g. for [products](https://github.com/getconversio/go-shopify/blob/e461b55bc05f9b6e038e02d2b3fb91e1dc2d6d17/product.go#L37).

**Review by dlebech — CHANGES_REQUESTED** [review]
Thank you for the PR :+1: :1st_place_medal:

Only a few tweaks to make :slightly_smiling_face:

File: customcollection.go (hunk 2 above)
Comment (jeremy-parr) [reply customcollection.go]: Docs are wrong on that one for some reason. I've thrown an example object below from the API.

```json
        {
            "id": (removed-int-id),
            "handle": "example-manual-collection",
            "title": "Example Manual Collection",
            "updated_at": "2018-02-06T01:48:03-05:00",
            "body_html": "Example body.",
            "published_at": "2018-02-06T01:42:43-05:00",
            "sort_order": "best-selling",
            "template_suffix": null,
            "published_scope": "web",
            "image": {
                "created_at": "2018-02-06T01:43:10-05:00",
                "width": 3000,
                "height": 2000,
                "src": "(removed image URL)"
            }
        },
```

**Comment by jeremy-parr** [comment]
Fixed up those changes from the other day, if you find anything else let me know.

**Review by dlebech — APPROVED** [review]

**Comment by dlebech** [comment]
Thanks again @fodawim ! 💯

File: customcollection.go (hunk 2 above)
Comment (dlebech) [reply customcollection.go]: Thanks for the example 👍

**Comment by jeremy-parr** [comment]
No problem, will probably be back with more PRs soon as I'm using it for a few internal things.

**Pull request merged**